`cmake . && make` fails in ExpressionAnalyzer.cpp

Installation on Debian 9 fails. See console log below:

````
roland@goms:~/Projekte/binaryen$ cmake . && make
-- The C compiler identification is GNU 6.3.0
-- The CXX compiler identification is GNU 6.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- No build type selected, default to Release
-- Building with -DBUILD_LLVM_DWARF
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Building with -std=c++14
-- Building with -Wall
-- Building with -Werror
-- Building with -Wextra
-- Building with -Wno-unused-parameter
-- Building with -fno-omit-frame-pointer
-- Building with -Wno-implicit-int-float-conversion
-- Building with -Wno-unknown-warning-option
-- Building with -Wswitch
-- Building with -fPIC
-- Building with -O2
-- Building with -UNDEBUG
-- Found PythonInterp: /usr/bin/python3.5 (found suitable version "3.5.3", minimum required is "3.5") 
-- Building libbinaryen as shared library.
-- Configuring done
-- Generating done
-- Build files have been written to: /home/roland/Projekte/binaryen
Scanning dependencies of target llvm_dwarf
[  0%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/Binary.cpp.o
[  1%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/ConvertUTF.cpp.o
[  1%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DataExtractor.cpp.o
[  2%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/Debug.cpp.o
[  2%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DJB.cpp.o
[  3%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/Dwarf.cpp.o
[  3%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/dwarf2yaml.cpp.o
[  4%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFAbbreviationDeclaration.cpp.o
[  4%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFAcceleratorTable.cpp.o
[  5%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFAddressRange.cpp.o
[  5%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFCompileUnit.cpp.o
[  6%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFContext.cpp.o
[  6%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFDataExtractor.cpp.o
[  7%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFDebugAbbrev.cpp.o
[  8%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFDebugAddr.cpp.o
[  8%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFDebugAranges.cpp.o
[  9%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFDebugArangeSet.cpp.o
[  9%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFDebugFrame.cpp.o
[ 10%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFDebugInfoEntry.cpp.o
[ 10%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFDebugLine.cpp.o
[ 11%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFDebugLoc.cpp.o
[ 11%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFDebugMacro.cpp.o
[ 12%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFDebugPubTable.cpp.o
[ 12%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFDebugRangeList.cpp.o
[ 13%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFDebugRnglists.cpp.o
[ 13%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFDie.cpp.o
[ 14%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFEmitter.cpp.o
[ 14%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFExpression.cpp.o
[ 15%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFFormValue.cpp.o
[ 16%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFGdbIndex.cpp.o
[ 16%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFListTable.cpp.o
[ 17%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFTypeUnit.cpp.o
[ 17%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFUnit.cpp.o
[ 18%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFUnitIndex.cpp.o
[ 18%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFVerifier.cpp.o
[ 19%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFVisitor.cpp.o
[ 19%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/DWARFYAML.cpp.o
[ 20%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/Error.cpp.o
[ 20%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/ErrorHandling.cpp.o
[ 21%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/FormatVariadic.cpp.o
[ 21%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/Hashing.cpp.o
[ 22%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/LEB128.cpp.o
[ 23%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/LineIterator.cpp.o
[ 23%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/MCRegisterInfo.cpp.o
[ 24%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/MD5.cpp.o
[ 24%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/MemoryBuffer.cpp.o
[ 25%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/NativeFormatting.cpp.o
[ 25%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/ObjectFile.cpp.o
[ 26%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/obj2yaml_Error.cpp.o
[ 26%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/Optional.cpp.o
[ 27%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/Path.cpp.o
[ 27%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/raw_ostream.cpp.o
[ 28%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/ScopedPrinter.cpp.o
[ 28%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/SmallVector.cpp.o
[ 29%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/SourceMgr.cpp.o
[ 29%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/StringMap.cpp.o
[ 30%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/StringRef.cpp.o
[ 31%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/SymbolicFile.cpp.o
[ 31%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/Twine.cpp.o
[ 32%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/UnicodeCaseFold.cpp.o
[ 32%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/WithColor.cpp.o
[ 33%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/YAMLParser.cpp.o
[ 33%] Building CXX object third_party/llvm-project/CMakeFiles/llvm_dwarf.dir/YAMLTraits.cpp.o
[ 33%] Built target llvm_dwarf
Scanning dependencies of target ir
[ 34%] Building CXX object src/ir/CMakeFiles/ir.dir/ExpressionAnalyzer.cpp.o
In file included from /home/roland/Projekte/binaryen/src/wasm-type.h:20:0,
                 from /home/roland/Projekte/binaryen/src/literal.h:28,
                 from /home/roland/Projekte/binaryen/src/wasm.h:34,
                 from /home/roland/Projekte/binaryen/src/wasm-traversal.h:32,
                 from /home/roland/Projekte/binaryen/src/ir/iteration.h:20,
                 from /home/roland/Projekte/binaryen/src/ir/ExpressionAnalyzer.cpp:17:
/home/roland/Projekte/binaryen/src/wasm-features.h:71:13: error: enclosing class of constexpr non-static member function ‘FeatureSet::operator uint32_t() const’ is not a literal type
   constexpr operator uint32_t() const { return features; }
             ^~~~~~~~
/home/roland/Projekte/binaryen/src/wasm-features.h:26:8: note: ‘FeatureSet’ is not literal because:
 struct FeatureSet {
        ^~~~~~~~~~
/home/roland/Projekte/binaryen/src/wasm-features.h:26:8: note:   ‘FeatureSet’ is not an aggregate, does not have a trivial default constructor, and has no constexpr constructor that is not a copy or move constructor
In file included from /home/roland/Projekte/binaryen/src/literal.h:28:0,
                 from /home/roland/Projekte/binaryen/src/wasm.h:34,
                 from /home/roland/Projekte/binaryen/src/wasm-traversal.h:32,
                 from /home/roland/Projekte/binaryen/src/ir/iteration.h:20,
                 from /home/roland/Projekte/binaryen/src/ir/ExpressionAnalyzer.cpp:17:
/home/roland/Projekte/binaryen/src/wasm-type.h:176:23: error: specialization of ‘template<class _Tp> struct std::hash’ in different namespace [-fpermissive]
 template<> class std::hash<wasm::Type> {
                       ^~~~~~~~~~~~~~~~
In file included from /usr/include/c++/6/bits/basic_string.h:5643:0,
                 from /usr/include/c++/6/string:52,
                 from /usr/include/c++/6/stdexcept:39,
                 from /usr/include/c++/6/array:39,
                 from /home/roland/Projekte/binaryen/src/support/small_vector.h:25,
                 from /home/roland/Projekte/binaryen/src/wasm-traversal.h:30,
                 from /home/roland/Projekte/binaryen/src/ir/iteration.h:20,
                 from /home/roland/Projekte/binaryen/src/ir/ExpressionAnalyzer.cpp:17:
/usr/include/c++/6/bits/functional_hash.h:58:12: error:   from definition of ‘template<class _Tp> struct std::hash’ [-fpermissive]
     struct hash;
            ^~~~
In file included from /home/roland/Projekte/binaryen/src/literal.h:28:0,
                 from /home/roland/Projekte/binaryen/src/wasm.h:34,
                 from /home/roland/Projekte/binaryen/src/wasm-traversal.h:32,
                 from /home/roland/Projekte/binaryen/src/ir/iteration.h:20,
                 from /home/roland/Projekte/binaryen/src/ir/ExpressionAnalyzer.cpp:17:
/home/roland/Projekte/binaryen/src/wasm-type.h:181:23: error: specialization of ‘template<class _Tp> struct std::hash’ in different namespace [-fpermissive]
 template<> class std::hash<wasm::Signature> {
                       ^~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/6/bits/basic_string.h:5643:0,
                 from /usr/include/c++/6/string:52,
                 from /usr/include/c++/6/stdexcept:39,
                 from /usr/include/c++/6/array:39,
                 from /home/roland/Projekte/binaryen/src/support/small_vector.h:25,
                 from /home/roland/Projekte/binaryen/src/wasm-traversal.h:30,
                 from /home/roland/Projekte/binaryen/src/ir/iteration.h:20,
                 from /home/roland/Projekte/binaryen/src/ir/ExpressionAnalyzer.cpp:17:
/usr/include/c++/6/bits/functional_hash.h:58:12: error:   from definition of ‘template<class _Tp> struct std::hash’ [-fpermissive]
     struct hash;
            ^~~~
cc1plus: error: unrecognized command line option ‘-Wno-unknown-warning-option’ [-Werror]
cc1plus: error: unrecognized command line option ‘-Wno-implicit-int-float-conversion’ [-Werror]
cc1plus: all warnings being treated as errors
src/ir/CMakeFiles/ir.dir/build.make:62: die Regel für Ziel „src/ir/CMakeFiles/ir.dir/ExpressionAnalyzer.cpp.o“ scheiterte
make[2]: *** [src/ir/CMakeFiles/ir.dir/ExpressionAnalyzer.cpp.o] Fehler 1
CMakeFiles/Makefile2:569: die Regel für Ziel „src/ir/CMakeFiles/ir.dir/all“ scheiterte
make[1]: *** [src/ir/CMakeFiles/ir.dir/all] Fehler 2
Makefile:138: die Regel für Ziel „all“ scheiterte
make: *** [all] Fehler 2

````

--- a/src/wasm-features.h
+++ b/src/wasm-features.h
@@ -68,7 +68,7 @@ struct FeatureSet {
 
   FeatureSet() : features(MVP) {}
   FeatureSet(uint32_t features) : features(features) {}
-  constexpr operator uint32_t() const { return features; }
+  operator uint32_t() const { return features; }
 
   bool isMVP() const { return features == MVP; }
   bool has(FeatureSet f) { return (features & f) == f; }

